sympy/functions/special/gamma_functions.py

```diff
@@ -657,22 +658,13 @@ def eval(cls, n, z):
             if z is S.NaN:
                 return S.NaN
             elif z.is_Rational:
-                # TODO actually *any* n/m can be done, but that is messy
-                lookup = {S(1)/2: -2*log(2) - S.EulerGamma,
-                          S(1)/3: -S.Pi/2/sqrt(3) - 3*log(3)/2 - S.EulerGamma,
-                          S(1)/4: -S.Pi/2 - 3*log(2) - S.EulerGamma,
-                          S(3)/4: -3*log(2) - S.EulerGamma + S.Pi/2,
-                          S(2)/3: -3*log(3)/2 + S.Pi/2/sqrt(3) - S.EulerGamma}
-                if z > 0:
-                    n = floor(z)
-                    z0 = z - n
-                    if z0 in lookup:
-                        return lookup[z0] + Add(*[1/(z0 + k) for k in range(n)])
-                elif z < 0:
-                    n = floor(1 - z)
-                    z0 = z + n
-                    if z0 in lookup:
-                        return lookup[z0] - Add(*[1/(z0 - 1 - k) for k in range(n)])
+
+                p, q = z.as_numer_denom()
+
+                # only expand for small denominators to avoid creating long expressions
```

Comment:
The comment looks good, but the *user* can't see the comment, how about we documented it as well may be that we need to use `expand_func` to expand beyond 5?

Reply:
Also how about if we think if introducing an argument in `cls` class method to expand in any case, even if the output expression is expected to be *long*. Through the argument the user will be able to control directly with the function expands itself or not.

Without the argument (in cls method), the user will need to explicitly call the `expand_func`.

I'm not sure, though the zen of python says:
```
There should be one-- and preferably only one --obvious way to do it.
```

But with this we'll have two ways to do it. WDYT?

Reply:
I like the idea of documenting the `eval` method better, and not only for this function.

The way the polygamma evaluation is implemented here is consistent with other SymPy functions where evaluations at specific values may be possible but not desirable. For example, `sin(pi/n)` returns evaluated if n is 2, 4, or 8, but not for 16, 32, 64... However, one can use `expand_trig` to get `sin(pi/16)`, etc to evaluate. 

So if instead of using `expand` we want a keyword argument saying "always expand", then for consistency the same should be done with sin, cos, tan, sec, cot, csc, and probably a bunch of other functions.  That looks like a wider change and should be discussed as a separate issue.

To repeat, I do support better explanation of what `eval` does and what `expand_` does. Currently some of docstrings don't explain much, for example, `asin` docstring basically says "read the code".

Reply:
> I like the idea of documenting the eval method better, and not only for this function.

I think I agree with you on this. Though there have been issues (for quite sometime infact) that the documentation isn't good as it can be. I think steps like these will be a way to move forward in that direction slowly.